src/core/action.ts

```diff
@@ -10,24 +11,37 @@ export class Action {
   readonly eventOptions: AddEventListenerOptions
   readonly identifier: string
   readonly methodName: string
+  readonly keyFilter: string
+  readonly schema: Schema
 
-  static forToken(token: Token) {
-    return new this(token.element, token.index, parseActionDescriptorString(token.content))
+  static forToken(token: Token, schema: Schema) {
+    return new this(token.element, token.index, parseActionDescriptorString(token.content), schema)
   }
 
-  constructor(element: Element, index: number, descriptor: Partial<ActionDescriptor>) {
+  constructor(element: Element, index: number, descriptor: Partial<ActionDescriptor>, schema: Schema) {
     this.element      = element
     this.index        = index
     this.eventTarget  = descriptor.eventTarget || element
     this.eventName    = descriptor.eventName || getDefaultEventNameForElement(element) || error("missing event name")
     this.eventOptions = descriptor.eventOptions || {}
     this.identifier   = descriptor.identifier || error("missing identifier")
     this.methodName   = descriptor.methodName || error("missing method name")
+    this.keyFilter    = descriptor.keyFilter || ''
+    this.schema = schema
   }
 
   toString() {
-    const eventNameSuffix = this.eventTargetName ? `@${this.eventTargetName}` : ""
-    return `${this.eventName}${eventNameSuffix}->${this.identifier}#${this.methodName}`
+    const eventFilter = this.keyFilter ? `.${this.keyFilter}` : ""
+    const eventTarget = this.eventTargetName ? `@${this.eventTargetName}` : ""
+    return `${this.eventName}${eventFilter}${eventTarget}->${this.identifier}#${this.methodName}`
+  }
+
+  isFilterTarget(key: string): boolean {
+    if (!(this.keyFilter && this.keyMappings[this.keyFilter])) {
+      return false;
+    }
+
+    return this.keyMappings[this.keyFilter] !== key
```

Comment:
Hey ! My first PR review on this repo :D 
Are you only checking if `keyFilter` exists in `keyMappings`? If yes, can this be simplified to

```suggestion
  isFilterTarget(key: string): boolean {
    if(this.keyMappings.hasOwnProperty(this.keyFilter)){
      return this.keyMappings[this.keyFilter] !== key;
    }
    return false;

```

Reply:
Thanks, I'll apply it if they decide they still need this PR. 😄